partiql-parser/src/main/antlr/PartiQL.g4

```diff
@@ -408,10 +408,28 @@ edgeWSpec
     ;
 
 edgeSpec
-    : BRACKET_LEFT symbolPrimitive? patternPartLabel? whereClause? BRACKET_RIGHT;
+    : BRACKET_LEFT symbolPrimitive? ( COLON labelSpec )? whereClause? BRACKET_RIGHT;
 
-patternPartLabel
-    : COLON symbolPrimitive;
+labelSpec
+    : labelSpec VERTBAR labelTerm        # LabelSpecOr
+    | labelTerm                          # LabelSpecTerm
+    ;
+
+labelTerm
+    : labelTerm AMPERSAND labelFactor    # LabelTermAnd
+    | labelFactor                        # LabelTermFactor
+    ;
+
+labelFactor
+    : BANG labelPrimary                  # LabelFactorNot
+    | labelPrimary                       # LabelFactorPrimary
+    ;
+
+labelPrimary
+    : symbolPrimitive                    # LabelPrimaryName
```

Comment:
From the GPML paper, I believe this would only be `IDENTIFIER`.

However, this would severely limit labels that share the name of reserved keywords. Just adding this comment here to make sure this was intentional. That being said -- it seems like label names are always case-sensitive whether they are quoted or not. Am I right in this assumption?

Reply:
Are you concerned that a label can only be a non-delimited/regular identifier? I did not go back to sift through the paper, but I would not expect it to bother about this issue. My expectation is that a primary label would be a general SQL identifier, i.e. both non-delimited and delimited being allowed. 

Here, I did not dwell on the matter much: it was a `symbolPrimitive` before [old Line 414 above], so it remains that. 
This should be resolved while dealing with #1122, which I have great hope of getting to soon.

Reply:
I guess it's more about case-sensitivity -- and retaining whether the name was delimited or not. The reason is that in `SimpleGraph` -- when matching a Graph against a LabelSpec, we check in a case-sensitive manner whether the graph's labels contain a LabelSpec. As this was the functionality prior to this PR, I can ignore this as out-of-scope until #1122 is addressed. Thanks!